### ISSUE TITLE
Item tooltip on tool changing

### ISSUE THREAD

**Issue by getsmp** [MovingBlocks/Terasology#1514]
It would be cool to see name of item you switched to when switching slot with 1...0 buttons.

**Comment by Cervator** [comment]
I think this is a good idea. We do have some tool tip action in some places, and even a module that enables them for things you're looking at.

It *might* be covered by existing suggestions threads or something like that, but will leave it here for now till we either finish this small piece or organize the tool tip ideas better together :-)

**Comment by Aniiram** [comment]
Hi! 
I am one of the Porto students. I want to ask you if my group can develop this feature, but i would like if possible to get more information about the module and the package related to it.

**Comment by Aniiram** [comment]
@Cervator

**Comment by Cervator** [comment]
Hi @Aniiram :-)

I checked on this thinking it might be out of date as we once did indeed not have tool tips for the tool bar. Which we do now. But maybe the original request here is triggering the tool tip simply when pressing 1, 2, etc, letting it appear without a mouse hover. Which seems like it could indeed be nice!

For finding some of the related code you could perhaps ping @Josharias who very recently did a small PR related to tool tips: #1994 - he's in our IRC channel quite often as well, though keep our different time zones in mind.

Other than that a quick search for "tooltip" in the code would probably help too. This would be in the engine for sure. Maybe input handling would relate, catching when the player presses the keys bound to the tool bar, and using that event to trigger a few second display of a tool tip for a particular button, resetting any recent such action so the tool tips don't end up overlapping.

**Comment by Aniiram** [comment]
Hi @Cervator! 
Sory to take so long to answer. I was analising the code and I think I've I already understand how to do. I will talk with @Josharias and probably forked with him :) Thanks for the help

**Comment by Cervator** [comment]
No worries on timing, glad to hear back from you! :-)

Good luck, hoping to see some nice code soon!

**Comment by Aniiram** [comment]
Yes @Cervator I'm trying. I already know what to change (at least I already have an ideia) but I want to know somwthing: what is this object: InteractionRegion. Can you help me?

**Comment by Cervator** [comment]
Hey @Aniiram. Rather than give you a poor answer myself (not very familiar with that code) I'd like to defer to @Josharias, @msteiger, or @skaldarnar. They're likely much more familiar with the area and able to give you a more useful answer :-)

**Comment by Josharias** [comment]
Aniiram and I chatted on IRC back and forth a bit about interaction regions, getting tooltip text, and general strategy.  I suggested doing something similar to the WorldlyTooltip module.  His goal is something like this with just the display name: https://youtu.be/Bpnzm9Wjz4k?t=23s

**Comment by Cervator** [comment]
:+1:

**Comment by Cervator** [comment]
Bump. Hey @Aniiram I noticed https://github.com/andrelago13/Terasology/pull/25 - are you getting close? was that supposed to go to MovingBlocks/Terasology? Any remaining questions we can badger @Josharias about to help you out? :-)

**Comment by Aniiram** [comment]
Hey @Cervator I am getting closer but I get stoped with an obstacle. I will try to talk with @Josharias this weekend. I couldn't earlier because I was handling school projects.

**Comment by Cervator** [comment]
That's fine and understandable, just wanted to make sure you weren't stuck without knowing what to do next or who to ask :-)

**Comment by Sabo2k** [comment]
I've been looking for some issues to contribute to and the implementation of this feature seems to have stopped. Can I help with this one in anyway? :)

**Comment by jdrueckert** [comment]
Hi @Sabo2k and welcome :wave: 
We're currently not so much in feature development mode unfortunately but more focusing on refactorings to improve the quality and stability of our codebase, so this one has relatively low priority, but there are other issues that you could help us with if you're not set on feature development.
Generally you're free to work on any issue you'd like (no need to ask on the issue whether you can work on it, just open a PR).
However, things that have lower priority for us will take longer to get reviewed and merged.

TL;DR: If you'd like to work on this, send in a PR and we'll see when we get to reviewing it, if you're open to help with refactorings, drop by our [Discord Server](https://discord.gg/terasology) and we can talk about a suitable task :slightly_smiling_face:
